### PR TITLE
Use "full" OpenJDK image 

### PR DESCRIPTION
**Pull request by cescoffier**
and switch back to run-java to start the JVM application

### CODE REVIEW

**Comment by gsmet** [comment]
Marking this ready for review to trigger CI.

**Review by gsmet — DISMISSED** [review]
We are waiting for @maxandersen 's green light on this one.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute [hunk 1]
```diff
@@ -20,16 +20,71 @@
 #
 # docker run -i --rm -p 8080:8080 quarkus/{project.artifact-id}-{type}
 #
+# This image uses the `run-java.sh` script to run the application.
```
Comment (maxandersen) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:23]: These docs are fine by now but really should consider putting this in a guide somewhere and link to it instead.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute [hunk 2]
```diff
@@ -20,16 +20,71 @@
...
 {#insert copy /}
 
 EXPOSE 8080
 USER 185
-
-ENTRYPOINT [ "java", "-jar", "/deployments/quarkus-run.jar" ]
+{#if java.version == '11'}
+ENV AB_JOLOKIA_OFF=""
+{/if}
+ENV JAVA_OPTS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
```
Comment (maxandersen) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:89]: JAVA_OPTIONS is standard Java vm env var.

What is the semantics of JAVA_OPTS here?

**Review by maxandersen — APPROVED** [review]
lgtm - I can't find fault in these changes; but I found issues in openshift and container image deploy that made it hard/impossible to 100% verify - but afaics this specific change is in a good enough state to merge. if issues it would also be present with the current one in main.

**Comment by jorsol** [comment]
I'm curious to know the reasoning about using the full OpenJDK image instead of the runtime... is the runtime image not compatible/supported with Quarkus?

On a side note, the [Migration-Guide-2.7](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.7) should also be updated.

**Comment by maxandersen** [comment]
It is. It's just that the runtime image was planned to have run-java by now to simplify /unify how debug and memory settings were handled. 

That unfortunately didn't make it in yet so instead of breaking "apis" for a period we moved to the full image. 

Intention is 100% to move to smaller image by default when we can. 

You can in your own customization of container build choose to just use the runtime image or some other image that has a java runtime.

**Pull request merged**